src/server/v2.0/handler/user.go

```diff
@@ -0,0 +1,485 @@
+		return u.SendError(ctx, err)
+	}
+	if err := u.ctl.SetCliSecret(ctx, uid, params.Secret.Secret); err != nil {
+		logger.Errorf("Failed to set CLI secret, error: %v", err)
+		return u.SendError(ctx, err)
+	}
+	return users.NewSetCliSecretOK()
+}
+
+func (u *usersAPI) CreateUser(ctx context.Context, params users.CreateUserParams) middleware.Responder {
+	logger := log.GetLogger(ctx)
+	if err := u.requireCreatable(ctx); err != nil {
+		return u.SendError(ctx, err)
+	}
+	if err := requireValidSecret(params.UserReq.Password); err != nil {
+		return u.SendError(ctx, err)
+	}
+	m := &usermodels.User{
+		Username: params.UserReq.Username,
+		Realname: params.UserReq.Realname,
+		Email:    params.UserReq.Email,
+		Comment:  params.UserReq.Comment,
+		Password: params.UserReq.Password,
+	}
+	if err := validateUserProfile(m); err != nil {
+		return u.SendError(ctx, err)
+	}
+	uid, err := u.ctl.Create(ctx, m)
+	if err != nil {
+		logger.Errorf("Failed to create user, error: %v", err)
+		return u.SendError(ctx, err)
+	}
+	location := fmt.Sprintf("%s/%d", strings.TrimSuffix(params.HTTPRequest.URL.Path, "/"), uid)
+	return users.NewCreateUserCreated().WithLocation(location)
+
+}
+
+func (u *usersAPI) ListUsers(ctx context.Context, params users.ListUsersParams) middleware.Responder {
+	if err := u.RequireSystemAccess(ctx, rbac.ActionList, userResource); err != nil {
+		return u.SendError(ctx, err)
+	}
+	query, err := u.BuildQuery(ctx, params.Q, params.Sort, params.Page, params.PageSize)
+	if err != nil {
+		return u.SendError(ctx, err)
+	}
+	values := params.HTTPRequest.URL.Query()
+	for _, k := range []string{"username", "email"} {
+		if v := values.Get(k); v != "" {
+			query.Keywords[k] = &q.FuzzyMatchValue{Value: v}
+		}
+	}
+	total, err := u.ctl.Count(ctx, query)
+	if err != nil {
+		return u.SendError(ctx, err)
+	}
+	ul, err := u.ctl.List(ctx, query)
+	if err != nil {
+		return u.SendError(ctx, err)
+	}
+	payload := make([]*models.UserResp, len(ul))
+	for i, u := range ul {
+		m := &model.User{
+			User: u,
+		}
+		payload[i] = m.ToUserResp()
+	}
+	return users.NewListUsersOK().
+		WithPayload(payload).
+		WithLink(u.Links(ctx, params.HTTPRequest.URL, total, query.PageNumber, query.PageSize).String()).
+		WithXTotalCount(total)
+}
+
+func (u *usersAPI) GetCurrentUserPermissions(ctx context.Context, params users.GetCurrentUserPermissionsParams) middleware.Responder {
+	if err := u.RequireAuthenticated(ctx); err != nil {
+		u.SendError(ctx, err)
+	}
+	scope := ""
+	if params.Scope != nil {
+		scope = *params.Scope
+	}
+	var policies []*types.Policy
+	sctx, _ := security.FromContext(ctx)
+	if ns, ok := types.NamespaceFromResource(rbac.Resource(scope)); ok {
+		for _, policy := range ns.GetPolicies() {
+			if sctx.Can(ctx, policy.Action, policy.Resource) {
+				policies = append(policies, policy)
+			}
+		}
+	}
+	var res []*models.Permission
+	relative := params.Relative != nil && *params.Relative
```

Comment:
`relative := lib.BoolValue(params. Relative)` ?

Reply:
Ah, thanks.  I'm not aware of this utility.